presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetastoreClosure.java

```diff
@@ -336,14 +336,14 @@ public void acquireTableWriteLock(HiveIdentity identity, String queryId, long tr
         delegate.acquireTableWriteLock(identity, queryId, transactionId, dbName, tableName, operation, isPartitioned);
     }
 
-    public void updateTableWriteId(HiveIdentity identity, String dbName, String tableName, long transactionId, long writeId, OptionalLong rowCountChange)
+    public void updateTableWriteId(HiveIdentity identity, String dbName, String tableName, long transactionId, long writeId, OptionalLong rowCountChange, EnvironmentContext context)
```

Comment:
Note that EnvironmentContext is thrift api-specific and in theory interface doesn't "know" it talks to a thrift metastore, so would be better to avoid putting this in here.

is it that the passed value is always `new EnvironmentContext()`?

Reply:
Good point. I just followed the pattern already being followed for alterTransactionalTable, but makes sense I will go ahead and update the same for both updateTableWriteId and alterPartitions. Will also approve your [change](https://github.com/prestosql/presto/pull/6338) once the CI tests pass. 

Yes presently its always new EnvironmentContext()